Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -127,12 +127,21 @@ def get_sampling_params_dict(self, params_json):
         params_dict = json.loads(params_json)
 
         # Special parsing for the supported sampling parameters
-        # TODO: Add more parameters if needed
-        float_keys = ["temperature", "top_p"]
+        bool_keys = ["ignore_eos", "skip_special_tokens", "use_beam_search"]
+        for k in bool_keys:
+            if k in params_dict:
+                params_dict[k] = bool(params_dict[k])
+
+        float_keys = ["frequency_penalty", "length_penalty", "presence_penalty", "temperature", "top_p"]
         for k in float_keys:
             if k in params_dict:
                 params_dict[k] = float(params_dict[k])
 
+        int_keys = ["best_of", "max_tokens", "n", "top_k"]
+        for k in int_keys:
+            if k in params_dict:
+                params_dict[k] = int(params_dict[k])
+
         return params_dict
```

Comment:
Nice addition. Can you double check the `request.parameters()` path works?

From what I can tell we're doing this for the non-input tensor path:
```
parameters = request.paramters() # dictionary
json.loads(parameters) # ERROR: expected string?
```

Reply:
Thanks! That path seems to work fine locally. Perhaps parameters is returning a string or json.loads can handle a dictionary input?

Reply:
I think `request.parameters()` is always a string

Reply:
Ah yes, you're right: https://github.com/triton-inference-server/python_backend/tree/main#inference-request-parameters. I always use it via json loads() without thinking.

Thanks for checking!